### PR TITLE
Store ca bundle in FederatedCluster instead of the cluster secret

### PR DESCRIPTION
**Pull request by marun**
Since the ca bundle for a member cluster doesn't need to be protected and will be optional for some auth configurations, it can be stored in an optional field on `FederatedCluster` instead of in the cluster secret.

### CODE REVIEW

**Comment by marun** [comment]
CI failure was due to #839

**Comment by marun** [comment]
Rebased and ready for review.

**Comment by yamt** [comment]
@marun 
fix-ca-for-k3s.sh script needs an update, right? (just asking)

**Comment by marun** [comment]
@yamt Added an update for the k3s script, PTAL.

File: pkg/kubefedctl/join.go [hunk 1]
```diff
@@ -834,9 +845,13 @@ func populateSecretInHostCluster(clusterClientset, hostClientset kubeclient.Inte
 	v1SecretResult, err := hostClientset.CoreV1().Secrets(namespace).Create(&v1Secret)
 	if err != nil {
 		klog.V(2).Infof("Could not create secret in host cluster: %v", err)
-		return nil, err
+		return nil, nil, err
 	}
 
+	// caBundle is optional so no error is suggested if it is not
```
Comment (pmorie) [inline pkg/kubefedctl/join.go]: Since this moved out into the spec, is this comment still meaningful?

Comment (pmorie) [reply pkg/kubefedctl/join.go]: Nevermind, misread.

**Comment by pmorie** [comment]
This is LGTM; I'll leave for a non-hatter to review/tag

**Comment by yamt** [comment]
@marun thank you. lgtm

**Comment by marun** [comment]
Rebased

**Comment by shashidharatd** [comment]
CI link is not correct.

**Comment by marun** [comment]
The helm update script doesn't seem to be working for me. :/

**Comment by shashidharatd** [comment]
> The helm update script doesn't seem to be working for me. :/

Have you moved your local repo copy to `$GOPATH/src/sigs.k8s.io/kubefed` ?

**Comment by marun** [comment]
> > The helm update script doesn't seem to be working for me. :/
> 
> Have you moved your local repo copy to `$GOPATH/src/sigs.k8s.io/kubefed` ?

Yes.  Code compiles fine, tests run.  When I run the update script, no worky - it doesn't update it.

**Comment by marun** [comment]
pebkac - I've been stuck using `scripts/sync-up-helm-script.sh` when I should be using `make generate`.

**Comment by marun** [comment]
The helm chart issue turned out to be stale content in `./config/crds`.  I've updated the sync script to clean that path before generating.

File: scripts/sync-up-helm-chart.sh [hunk 2]
```diff
@@ -34,6 +34,9 @@ then
   exit 1
 fi
 
+# Remove existing generated crds to ensure that stale content doesn't linger.
+rm -f ./config/crds/*.yaml
```
Comment (irfanurrehman) [inline scripts/sync-up-helm-chart.sh]: Isn't this supposed to be a local fix?

**Pull request merged**